### ISSUE TITLE
False positives for NUnit2046 on non-array-like types

### ISSUE THREAD

**Issue by smkanadl** [nunit/nunit.analyzers#459]
I really like the `Has.Count` and `Has.Length` syntax, so I use it a lot.

In some projects, I have types that similar to the following example
```
class Pipe
{
    double Diameter { get; set; }

    double Length { get; set; }
}

Assert.That(pipe.Diameter, Is.EqualTo(1));
Assert.That(pipe.Length, Is.EqualTo(2)); // NUnit2046 Use Assert.That(<collection>, Has.Length.EqualTo(<value>)
```

I know it is not a big deal, but I was wondering if the analyzer could check for the actual type instead of raising the suggestion for all Asserts including a `Length` (or `Count`) property. Not all types having such a property are actually Arrays, Lists or Collections :-)

**Comment by mikkelbu** [comment]
I think it makes sense to improve the logic in UseCollectionConstraintAnalyzer.cs so that we reduce the number of false positives.

**Comment by manfred-brands** [comment]
I think this is covered by #445. Can you try the pre-release package at [myget.org](https://www.myget.org/feed/nunit-analyzers/package/nuget/NUnit.Analyzers/3.4.0-dev-00952)?

**Comment by manfred-brands** [comment]
@smkanadl You can try to package from this PR: https://ci.appveyor.com/project/mikkelbu/nunit-analyzers/builds/43651598/artifacts

**Comment by smkanadl** [comment]
@manfred-brands All warnings gone. Test passed. Thank you very much!
